src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -55,6 +54,51 @@ public static Task StopAsync(this IWebHost host, TimeSpan timeout)
         }
 
         /// <summary>
+        /// Block the calling thread until shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static void WaitForShutdown(this IWebHost host)
+        {
+            host.WaitForShutdownAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Block the calling thread until shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static void WaitForShutdown(this IWebHost host, CancellationToken token)
+        {
+            host.WaitForShutdownAsync(token).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static Task WaitForShutdownAsync(this IWebHost host)
+        {
+            return host.WaitForShutdownAsync(CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static async Task WaitForShutdownAsync(this IWebHost host, CancellationToken token)
```

Comment:
To simplify the rest of this: version A = RunAsync(), B = RunAsync(CancellationToken ct = default(CancellationToken))

It makes sense to make this change for `WaitForShutdownAsync` but why do we need to change `RunAsync`? The rationale for this pattern, to my understanding, is so that we don't need to create the overload without the cancellation token but for RunAsync, the behaviour is fundamentally different between A and B. Now, it is ambiguous what RunAsync() means. Was the intention to use A or B but with the default value?

I was surprised this compiles at all since RunAsync() could potentially resolve to either A or B but the compiler seems to prefer A. I also verified that if A is removed, RunAsync() will resolve to B but will use the default value. But given the compiler prefers A, B will never be invoked with the default value so I don't see why that's necessary to add.

If the goal is to combine the implementation of A and B into one then it will be hard to distinguish between RunAsync() and RunAsync(CancellationToken.None), which today behaves differently.